aws/signer/v4/eventstream.go

```diff
@@ -0,0 +1,63 @@
+package v4
+
+import (
+	"encoding/hex"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+)
+
+type credentialValueProvider interface {
+	Get() (credentials.Value, error)
+}
+
+// EventStreamSigner implements signing of event stream encoded payloads
+type EventStreamSigner struct {
+	region  string
+	service string
+
+	credentials credentialValueProvider
+
+	prevSig []byte
+}
+
+// NewEventStreamSigner creates a SigV4 signer used to sign Event Stream encoded messages
+func NewEventStreamSigner(region, service string, seedSignature []byte, credentials *credentials.Credentials) *EventStreamSigner {
```

Comment:
Minor preference, but what do you think about making this more generic to StreamSigning instead of EventStream?

Reply:
Working on getting clarification on this out-of-band.